cassiopeia/solver/solver_utilities.py

```diff
@@ -18,6 +18,14 @@ class PriorTransformationError(Exception):
     pass
 
 
+def node_name_generator() -> Generator[str, None, None]:
+    """Generates unique node names for building the reconstructed tree."""
+    i = 0
+    while True:
+        yield "cassiopeia_internal_node_" + str(i)
+        i += 1
+
+
```

Comment:
Some questions on behavior -- 

(1) Every time you call this function does this create a _new_ generator? As in there won't be contamination between generators being used in separate functions right?

(2) What else do you do to ensure that you aren't creating duplicate nodes in the CassiopeiaTree? For example, let's consider a pathological example where a character matrix entries starting with `cassiopeia_internal_node` -- how do we handle this? Is there a more unique identifier that we can use? One possibility is to hash the timestamp and use this as the identifier before the unique integer. Thoughts?